### ISSUE TITLE
mid and high frequencies missing

### ISSUE THREAD

**Issue by fattyyyyyyyyyy** [starrytong/SCNet#9]
I've been training a six-track separation model, but I've encountered an issue where the "other," "piano," and "guitar" tracks tend to lose mid-to-high frequencies. The cutoff point corresponds to the crossover frequency in the band configuration. For example, with a band configuration of {'low': { 'SR': .175, 'stride': 1, 'kernel': 3 },'mid': { 'SR': .462, 'stride': 3, 'kernel': 4 },'high': {'SR': .433, 'stride': 16, 'kernel': 16 }}, frequencies above 3858Hz (22050 * 0.175) or 12502Hz(22050 * 0.567)are missing. After investigation, I found that this issue is influenced by both the data and the model. My private data for these tracks often has sparse frequency content and long silent segments. However, using the same data to train models like Demucs or BSRNN does not result in this frequency loss. Have anyone ever encountered a similar situation before?
<img width="994" alt="image" src="https://github.com/user-attachments/assets/cef0308c-d11b-4341-8859-897c766a6e0d">
<img width="985" alt="image" src="https://github.com/user-attachments/assets/338a5cd7-d710-448d-922e-cee6c1d7db3f">

**Comment by starrytong** [comment]
Perhaps you could provide the mixture of this split sample

**Comment by fattyyyyyyyyyy** [comment]
> Perhaps you could provide the mixture of this sample
[courage.mp3.zip](https://github.com/user-attachments/files/16262528/courage.mp3.zip)
This mixture is just a random pop song from the internet.

**Comment by starrytong** [comment]
I used 4-tracks separation model, the 'other' track is ok. If 'guitar' and 'piano'  track have this problem, it might be due to insufficient training data,  causing the model focuses on low-frequency.

There are two ways to mitigate this. 1. Using waveform loss instead of spectrum loss 2. Retraining a model for 'guitar' and 'piano' and adjust its SR.

**Comment by fattyyyyyyyyyy** [comment]
Thanks for the reply. My loss is waveform loss + spectrum loss, without any weight. I'll try to adjust the ratio of waveform loss.

**Comment by crlandsc** [comment]
I found that I had an improvement of high frequency separation fidelity when using the log-WMSE loss, which I adapted for PyTorch here: https://github.com/crlandsc/torch-log-wmse

It is basically a frequency weighted loss that can handle digital silence.
